server/src/main/java/org/jetbrains/bsp/bazel/server/diagnostics/DiagnosticBspMapper.kt

```diff
@@ -12,24 +12,22 @@ import ch.epfl.scala.bsp4j.Range as BspRange
 
 class DiagnosticBspMapper(private val bazelInfo: BazelInfo) {
 
-  fun diagnostics(parsedDiagnostics: List<Diagnostic>): List<PublishDiagnosticsParams> =
-      sequence {
-        val grouped = parsedDiagnostics.groupBy { it.fileLocation }
-        for ((location, diagnostics) in grouped) {
-          yield(createParams(location, diagnostics))
-        }
-      }.toList()
-
-  private fun createParams(
-      rawFileLocation: String,
-      fileDiagnostics: List<Diagnostic>
-  ): PublishDiagnosticsParams {
-    val fileLocation = TextDocumentIdentifier(toAbsoluteUri(rawFileLocation))
-    val targetId = BuildTargetIdentifier(fileDiagnostics[0].targetLabel ?: "")
-    val diagnostics = fileDiagnostics.map { createDiagnostic(it) }
-    return PublishDiagnosticsParams(fileLocation, targetId, diagnostics, false)
+  fun createDiagnostics(diagnostics: List<Diagnostic>, target: String): List<PublishDiagnosticsParams> {
+    val ident = BuildTargetIdentifier(target)
+    if (diagnostics.isEmpty()) {
+      return listOf(PublishDiagnosticsParams(TextDocumentIdentifier("unknown"), ident, listOf(), false))
```

Comment:
hm?

Reply:
i mean why we would like to send such diagnostics

Reply:
oh, do you mean we should just return an empty list here instead?  I figured this way we'd at least know a target failed even if we can't figure out why.

Reply:
I would keep the change, we will observe it in practice, if it even displays and if is useful and then decide.

Reply:
okey, then imo it should be inside `AllCatchParser`, not in the mapper

Reply:
I split the difference here and added it back to the DiagnosticsService, it's a little tricky to plumb it into the AllCatchParser since it'd need the target name there.

Reply:
>I split the difference here and added it back to the DiagnosticsService, it's a little tricky to plumb it into the AllCatchParser since it'd need the target name there.

we could pass target name to parsers, then we would be able to set target name for java compiler

Reply:
alright I refactored this a little more.